library/src/main/java/com/google/android/exoplayer/extractor/flv/ScriptTagPayloadReader.java

```diff
@@ -0,0 +1,258 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer.extractor.flv;
+
+import com.google.android.exoplayer.C;
+import com.google.android.exoplayer.MediaFormat;
+import com.google.android.exoplayer.extractor.TrackOutput;
+import com.google.android.exoplayer.util.MimeTypes;
+import com.google.android.exoplayer.util.ParsableByteArray;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Parses Script Data tags from an FLV stream and extracts metadata information.
+ */
+final class ScriptTagPayloadReader extends TagPayloadReader {
+
+  // AMF object types
+  private static final int AMF_TYPE_UNKNOWN = -1;
+  private static final int AMF_TYPE_NUMBER = 0;
+  private static final int AMF_TYPE_BOOLEAN = 1;
+  private static final int AMF_TYPE_STRING = 2;
+  private static final int AMF_TYPE_OBJECT = 3;
+  private static final int AMF_TYPE_MOVIE_CLIP = 4;
+  private static final int AMF_TYPE_NULL = 5;
+  private static final int AMF_TYPE_UNDEFINED = 6;
+  private static final int AMF_TYPE_REFERENCE = 7;
+  private static final int AMF_TYPE_ECMA_ARRAY = 8;
+  private static final int AMF_TYPE_END_MARKER = 9;
+  private static final int AMF_TYPE_STRICT_ARRAY = 10;
+  private static final int AMF_TYPE_DATE = 11;
+  private static final int AMF_TYPE_LONG_STRING = 12;
+
+  public long startTime = C.UNKNOWN_TIME_US;
+  public float frameRate;
+  public float videoDataRate;
+  public float audioDataRate;
+  public int height;
+  public int width;
+  public boolean canSeekOnTime;
+  public String httpHostHeader;
+
+  /**
+   * @param output A {@link TrackOutput} to which samples should be written.
+   */
+  public ScriptTagPayloadReader(TrackOutput output) {
+    super(output);
+
+    output.format(MediaFormat.createTextFormat(MimeTypes.APPLICATION_AMF, MediaFormat.NO_VALUE,
+        durationUs, null));
+  }
+
+  @Override
+  public void seek() {
+
+  }
+
+  @Override
+  protected void parseHeader(ParsableByteArray data) throws UnsupportedTrack {
+
+  }
+
+  @Override
+  protected void parsePayload(ParsableByteArray data, long timeUs) {
+    // TODO: Just parsing metadata to extract information about the stream but
+    // don't generating samples as we don't have any renderer that is going to consume
+    // them
+    Object messageName = readAMFData(data, AMF_TYPE_UNKNOWN);
+    Object obj = readAMFData(data, AMF_TYPE_UNKNOWN);
+
+    if (obj instanceof Map) {
+      Map<String, Object> extractedMetadata = (Map<String, Object>) obj;
+      for (Map.Entry<String, Object> entry : extractedMetadata.entrySet()) {
+        if (entry.getValue() == null) {
+          continue;
+        }
+
+        switch (entry.getKey()) {
+          case "duration":
+            this.durationUs = (long)(C.MICROS_PER_SECOND * (Double)(entry.getValue()));
+            break;
```

Comment:
Duration in an FLV's metadata, is in seconds. http://help.adobe.com/en_US/AS2LCR/Flash_10.0/help.html?content=00002413.html